### ISSUE TITLE
Scatter log softmax operation

### ISSUE THREAD

**Issue by mallamanis** [rusty1s/pytorch_scatter#76]
Would you be open for me to contribute a `scatter_log_softmax` operation in the library? It's a few lines of code using the existing APIs that'd be nice to have in this library, rather than somewhere else.

Adding `scatter_logsumexp` of #71 would also seem a reasonable step if you're okay with `scatter_log_softmax`.

**Comment by rusty1s** [comment]
The main problem with this is that this is not really a `scatter_*` operation, it just uses `scatter_*` ops under the hood. I am generally open to it, but we need to distinguish those kinds of ops.

**Comment by mallamanis** [comment]
Certainly agree that it's not an "primitive" `scatter_*` operation. It's nevertheless an op that I find myself needing relatively frequently that it'd like to encapsulate in _some_ library to reuse.  One suggestion is to put such functions in a `pytorch_scatter.composite` sub-package.

If `pytorch-scatter` is or isn't the right place for this, it's up to you as the owner to decide :) Let me know.

**Comment by rusty1s** [comment]
I like this idea. We can start by adding `scatter_softmax` and `scatter_log_softmax` to `torch_scatter.composite` and adding `scatter_logsumexp` to the main package.

**Comment by mallamanis** [comment]
Great! And thanks for your prompt responses. I aim to get some time to work on a PR later today or tomorrow morning.
